README.md

```diff
@@ -8,40 +8,141 @@
 [![Get invited](https://slack.developers.italia.it/badge.svg)](https://slack.developers.italia.it/)
 [![Join the #spid openid](https://img.shields.io/badge/Slack%20channel-%23spid%20openid-blue.svg)](https://developersitalia.slack.com/archives/C7E85ED1N/)
 
-The SPID/CIE OIDC Federation Relying Party, written in C# for AspNetCore
 > ⚠️ __This project is a WiP, the first stable release for production use will be the v1.0.0.__
 
+The SPID/CIE OIDC Federation Relying Party SDK, written in C# for AspNetCore. This is a custom implementation of an AspNetCore RemoteAuthenticationHandler for the OpenIdConnect profiles of SPID and CIE.
+
 ## Summary
 
 * [Features](#features)
 * [Setup](#setup)
-* [Docker compose](#docker-compose)
-* [Usage](#usage)
+* [Extensions](#extensions)
+* [Sample WebApp](#sample-webapp)
 * [Contribute](#contribute)
     * [Contribute as end user](#contribute-as-end-user)
     * [Contribute as developer](#contribute-as-developer)
 * [Implementations notes](#implementation-notes)
 * [License and Authors](#license-and-authors)
 
-## Setup
-
-> TODO: WiP
-
-## Docker compose
+## Features
+The purpose of this project is to provide a simple and immediate tool to integrate, in a WebApp developed with AspNetCore MVC, the authentication services of SPID and CIE, automating the login/logout flows, the management of the OIDC-Core/OIDC-Federation protocols and their security profiles, and simplify the development activities.
+Inside the repository there is both the library code (Spid.Cie.OIDC.AspNetCore), and a demo web app (Spid.Cie.AspNetCore.WebApp) which shows a sample integration of the library into an AspNetCore MVC webapp.
 
-> TODO: Not available until v1.0.0 release
-
-## Usage
+## Setup
 
-> TODO: WiP
+The library is distributed in the form of a NuGet package, which can be installed via the command
+
+`Install-Package Spid.Cie.OIDC.AspNetCore`
+
+At this point it is sufficient, inside the `Startup.cs`, to add the following lines:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+    {
+        var certificate = new X509Certificate2("wwwroot/certificates/ComuneVigata-SPID.pfx", "P@ssW0rd!");
+
+        services.AddControllersWithViews();
+        services
+            .AddAuthentication(o =>
+            {
+                o.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                o.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                o.DefaultChallengeScheme = SpidCieConst.AuthenticationScheme;
+            })
+            .AddCookie()
+            .AddSpidCieOIDC(o =>
+            {
+                o.LoadFromConfiguration(Configuration);
+
+                o.RequestRefreshToken = true;
+                o.SpidOPs.Add("http://127.0.0.1:8000/oidc/op/"); // Example SPID OP Url
```

Comment:
For a good dockerimage strategy, is it possibile to retrieve these values from ENV or do we have in the documentation a way to dynamically configure these?

Reply:
Does these parameter be compiled in C# or is this a configuration file that we can copy in a docker volume before starting the container?

Reply:
It's possible to read them from an external settings file, I'll add the feature